ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -255,7 +255,7 @@ getParsedModuleRule recorder =
   define (cmapWithPrio LogShake recorder) $ \GetParsedModule file -> do
     ModSummaryResult{msrModSummary = ms'} <- use_ GetModSummary file
     sess <- use_ GhcSession file
-    let hsc = hscEnv sess
+    (ms', hsc) <- liftIO $ initPlugins (hscEnv sess) ms'
```

Comment:
It occurs to me that in order to always initialise plugins consistently, the initialisation should be done in the `ModSummary` rule, the  initialized `HscEnv` should be stored in a field of `ModSummaryResult`, and all rules downstream should use it instead of asking for a `GhcSession`. 

This refactoring (or something of the sort) will prevent a bunch of future bugs.